File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/ConnectNodeSetImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the result setting a node connection
 */
class ConnectNodeSetImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    /**
     * edges removed before setting the new edge
     */
    fun removedEdges(config: EdgeKeyOne.() -> Unit) = 
        add { "removedEdges{ " + EdgeKeyOne().apply(config).render(it) + "}"}


    /**
     * edge created by this request
     */
    fun newEdge(config: Edge.() -> Unit) = 
        add { "newEdge{ " + Edge().apply(config).render(it) + "}"}

}

File: gaia-sdk-javascript/src/graphql/request/type/ConnectNodeSetImpulse.ts

import {EdgeKeyOne} from "./EdgeKeyOne";
import {Edge} from "./Edge";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result setting a node connection
 */
export class ConnectNodeSetImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "ConnectNodeSetImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * edges removed before setting the new edge
     */
    public removedEdges = (config: (_:EdgeKeyOne) => void) => this.push((registry) => {
        const entity = new EdgeKeyOne();
        config(entity);
        return "removedEdges { " + entity.render(registry) + " }";
    });

    /**
     * edge created by this request
     */
    public newEdge = (config: (_:Edge) => void) => this.push((registry) => {
        const entity = new Edge();
        config(entity);
        return "newEdge { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

